pkg/config/parameters.go

```diff
@@ -89,6 +90,88 @@ func (a AccessLogType) Validate() error {
 const EnvoyAccessLog AccessLogType = "envoy"
 const JSONAccessLog AccessLogType = "json"
 
+type AccessLogFields []string
```

Comment:
Why don't you make this a `map[string]string` so that it can be fully denormalized in the config? i.e, after validation, break the config down into key/value pairs so that the caller in `internal/envoy/v2/accesslog.go` doesn't need to do it.

Reply:
We were thinking about that, but because this type is used for a [YAML config parser](https://github.com/projectcontour/contour/pull/3059/files/c007f5c3b7e6ad3edb1c617a57f945ded2e9490f#diff-a324876cd0bc5df1659faac7631005ed9d83b9445c6720142db3c8d2d3e83e56R374) we cannot change it as it will break the current API. One option is to save it to a different field and then copy to this fields as a `map[string]string`

Reply:
> We were thinking about that, but because this type is used for a [YAML config parser](https://github.com/projectcontour/contour/pull/3059/files/c007f5c3b7e6ad3edb1c617a57f945ded2e9490f#diff-a324876cd0bc5df1659faac7631005ed9d83b9445c6720142db3c8d2d3e83e56R374) we cannot change it as it will break the current API.

Oh good point, I had blanked that out. WDYT about adding a method to to the conversion:
```Go
func (a *AccessLogFields) AsFieldMap() map[string]string {
...
}
```

Reply:
Done